packages/gravity/blade-runner/src/plan/interface.ts

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type ReplicantParams, type ReplicantRuntimeParams } from './spec';
+
+export interface CommonTestEnv {
+  syncBarrier(key: string, amount: number): Promise<void>;
+  syncData<T>(key: string, amount: number, data?: T): Promise<T[]>;
+}
+
+export interface ReplicantEnv extends CommonTestEnv {}
+
+export interface SchedulerEnv<S> extends CommonTestEnv {
+  spawn<T>(brain: ReplicantBrain<T>, runtime: ReplicantRuntimeParams): Promise<Replicant<T, S>>;
+}
+
+export type ReplicantBrain<T> = { new (): T };
+
+export interface Replicant<T, S> {
+  brain: RpcHandle<T>;
+  kill(signal?: NodeJS.Signals | number): void;
+  params: ReplicantParams<S>;
+}
+
+export type RpcHandle<T> = {
```

Comment:
add short jsdoc explaning that this type extracts all methods from T and makes them async